File: banchus-app/src/main/java/pe/nanamochi/banchus/protocol/handlers/PongHandler.java
package pe.nanamochi.banchus.protocol.handlers;

import java.io.ByteArrayOutputStream;
import org.springframework.stereotype.Component;
import pe.nanamochi.banchus.core.Packets;
import pe.nanamochi.banchus.database.entity.Session;
import pe.nanamochi.banchus.packets.client.PongPacket;
import pe.nanamochi.banchus.protocol.AbstractPacketHandler;
import pe.nanamochi.banchus.protocol.HandleClientPacket;

@Component
@HandleClientPacket(value = Packets.OSU_PONG, checkForRestriction = true)
public class PongHandler extends AbstractPacketHandler<PongPacket> {
  @Override
  public void handle(PongPacket packet, Session session, ByteArrayOutputStream responseStream) {}
}

File: banchus-app/src/main/kotlin/pe/nanamochi/banchus/handlers/PongHandler.kt
package pe.nanamochi.banchus.handlers

import java.io.ByteArrayOutputStream
import org.springframework.stereotype.Component
import pe.nanamochi.banchus.core.PacketType
import pe.nanamochi.banchus.database.entity.Session
import pe.nanamochi.banchus.infrastructure.protocol.HandleClientPacket
import pe.nanamochi.banchus.packets.client.PongPacket
import pe.nanamochi.banchus.protocol.AbstractPacketHandler

@Component
@HandleClientPacket(type = PacketType.OSU_PONG, checkForRestriction = true)
class PongHandler : AbstractPacketHandler<PongPacket>(PacketType.OSU_PONG) {

    override fun handle(
        packet: PongPacket,
        session: Session,
        responseStream: ByteArrayOutputStream,
    ) {}
}
